src/sync/treiber_stack.rs

```diff
@@ -49,6 +49,20 @@ impl<T> TreiberStack<T> {
         self.try_pop()
     }
 
+    /// Swaps a stack with a new one.  Can be used to take all the
+    /// elements off the stack at once.
+    ///
+    /// Wait-free
+    pub fn swap(&self, new: TreiberStack<T>) -> TreiberStack<T> {
```

Comment:
I chose to do it similar to the swap method for AtomicBool and such. IMO a signature like `fn function(&x: A, &mut y: A);` is a bit ugly and confusing. I don't have a strong opinion either way though.